components/merkledb/src/values.rs

```diff
@@ -137,6 +179,101 @@ impl BinaryValue for () {
     }
 }
 
+impl<T1> BinaryValue for (T1,)
+where
+    T1: BinaryValue,
+{
+    fn to_bytes(&self) -> Vec<u8> {
+        self.0.to_bytes()
+    }
+
+    fn from_bytes(bytes: Cow<[u8]>) -> Result<Self, failure::Error> {
+        Ok((T1::from_bytes(bytes)?,))
+    }
+}
+
+impl<T1> ObjectHash for (T1,)
+where
+    T1: BinaryValue,
+{
+    fn object_hash(&self) -> Hash {
+        exonum_crypto::hash(&self.to_bytes())
+    }
+}
+
+/// Converts vector of byte vectors into flat byte vector. Also keeps their
+/// sizes as converted usize values and writes them before the vectors
+/// themselves to the output vector to be able to convert it back
+fn chunks_to_bytes(chunks: &Vec<Vec<u8>>) -> Vec<u8> {
```

Comment:
1. `chunks: &[Vec<u8>]` would be more idiomatic.
2. I think the implementation using [`Vec::extend_from_slice`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extend_from_slice) (and computing the needed capacity of the output vector) would be significantly faster since it would use memory copying primitives.